### PR TITLE
Don't allow circular references within stack variables

### PR DESCRIPTION
**Pull request by mhahn**
Fixes: https://github.com/remind101/stacker/issues/228

### CODE REVIEW

File: stacker/stack.py [hunk 1]
```diff
@@ -114,6 +114,11 @@ def requires(self):
         for lookup in self.lookups:
             if lookup.type == OUTPUT_LOOKUP_TYPE_NAME:
                 d = deconstruct(lookup.input)
+                if d.stack_name == self.name:
+                    message = (
+                        "Lookup: \"%s\" is a circular reference to: %s"
```
Comment (phobologic) [inline stacker/stack.py]: Can we provide the stack & variable that this lookup happens in as part of this error?  Might make it easier to chase down what is actually broken in some cases.  Especially with anchors and what not.

Comment (mhahn) [inline stacker/stack.py]: Yea, error message now looks like this:

```
Variable Var1 in stack vpc.1 has a ciruclar reference within lookup: vpc.1::FakeOutput
```

File: stacker/stack.py [hunk 2]
```diff
@@ -111,11 +111,22 @@ def requires(self):
                 requires.add(stack_fqn)
 
         # Add any dependencies based on output lookups
-        for lookup in self.lookups:
-            if lookup.type == OUTPUT_LOOKUP_TYPE_NAME:
-                d = deconstruct(lookup.input)
-                stack_fqn = self.context.get_fqn(d.stack_name)
-                requires.add(stack_fqn)
+        for variable in self.variables:
```
Comment (mhahn) [inline stacker/stack.py]: in stacker v1.0 this is the only code within this function, so i don't think it needs to be broken out more

Comment (phobologic) [inline stacker/stack.py]: Cool - is there any reason to do away with `stack.lookups`?

Comment (mhahn) [inline stacker/stack.py]: yea we can kill them.

**Pull request merged**